lib/programmemory.cpp

```diff
@@ -34,6 +34,15 @@ bool ProgramMemory::getTokValue(nonneg int varid, const Token** result) const
     return found;
 }
 
+bool ProgramMemory::getContainerSizeValue(nonneg int varid, MathLib::bigint* result) const
```

Comment:
I don't want that you change anything now. But would it maybe be a bit more elegant to return a `std::optional` from such methods? I am not sure if Visual Studio 2013 / GCC 4.6 had that but it's possible.

Reply:
> But would it maybe be a bit more elegant to return a std::optional from such methods?

Yea probably. It could also use a `std::vector` like what I did for the `evaluate` function in `ForwardAnalyzer`.

Reply:
https://en.cppreference.com/w/cpp/compiler_support indicates that C++17 features are quite rare within VS 2013...

Reply:
Ah thanks.. then `std::optional` will have to wait a bit. :-(